packages/contentstack-bulk-publish/src/producer/publish-entries.js

```diff
@@ -37,14 +42,31 @@ async function getEntries(stack, contentType, locale, bulkPublish, environments,
       .then(async (entriesResponse) => {
         skipCount += entriesResponse.items.length;
         let entries = entriesResponse.items;
-        for (let index = 0; index < entriesResponse.items.length; index++) {
+
+        for (let index = 0; index < entries.length; index++) {
+          let variants = [];
+
+          if (variantsFlag) {
+            const variantsEntriesResponse = await stack
+              .contentType(contentType)
+              .entry(entries[index].uid)
+              .variants()
+              .query(queryParams)
+              .find();
+
+            variants = variantsEntriesResponse.items.map(entry => ({
+              uid: entry.variants_uid,
+            }));
+          }
+
           if (bulkPublish) {
             if (bulkPublishSet.length < 10) {
               bulkPublishSet.push({
                 uid: entries[index].uid,
                 content_type: contentType,
                 locale,
                 publish_details: entries[index].publish_details || [],
+                variants: variants
```

Comment:
@sunil-lakshman , [Minor], we can directly use **variants** as param instead of **variants:variants**

Reply:
@aman19K The payload is constructed explicitly with the required values to ensure clarity when sending it for publishing variant entries. This approach maintains consistency in the payload structure, especially with multiple parameters.